src/storage-operators/src/s3_oneshot_sink.rs

```diff
@@ -74,18 +73,32 @@ pub fn copy_to<G, F>(
         s3_key_manager.clone(),
     );
 
-    let completion_stream = render_upload_operator(
-        scope.clone(),
-        connection_context.clone(),
-        aws_connection.clone(),
-        connection_id,
-        connection_details,
-        sink_id,
-        input_collection,
-        err_collection,
-        up_to,
-        start_stream,
-    );
+    let completion_stream = match connection_details.format {
+        S3SinkFormat::PgCopy(_) => render_upload_operator::<G, pgcopy::PgCopyUploader>(
+            scope.clone(),
+            connection_context.clone(),
+            aws_connection.clone(),
+            connection_id,
+            connection_details,
+            sink_id,
+            input_collection,
+            err_collection,
+            up_to,
+            start_stream,
+        ),
+        S3SinkFormat::Parquet => render_upload_operator::<G, parquet::ParquetUploader>(
+            scope.clone(),
+            connection_context.clone(),
+            aws_connection.clone(),
+            connection_id,
+            connection_details,
+            sink_id,
+            input_collection,
+            err_collection,
+            up_to,
+            start_stream,
+        ),
```

Comment:
this copying being required for monomorphization is the worst part of rust generics haha